pkg/distro/rhel9/images.go

```diff
@@ -698,3 +708,25 @@ func initialSetupKickstart() *fsnode.File {
 	}
 	return file
 }
+
+// fixes: https://github.com/osbuild/images/issues/352
+// Creates a unit file that create mountpoints if it doesnot exits.
+// This ensures that the filesystem created using image builder continues to work after upgrade.
+func createFilesystemMounpointService(customizations *blueprint.Customizations) *fsnode.File {
+	mntpnts := customizations.GetFilesystems()
+	if mntpnts != nil {
+		customServiceUnit := "[Unit]\nDescription=Create mountpoints if does not exists\nDefaultDependencies=no\n"
+		customService := "[Service]\nType=oneshot\nRemainAfterExit=yes\nExecStartPre=chattr -i /\nExecStopPost=chattr +i /\nExecStart=mkdir -p"
```

Comment:
> One thing to note is this hack will break when switching to composefs. Ultimately then a use case that wants this would need to enable root.transient instead.

I have been pondering on this from: https://gitlab.com/CentOS/automotive/sample-images/-/merge_requests/451

cc @runcom @7flying 

Adding a warning seems to be feasible considering ITM's dead line.
and  following up with a spike on root.transient.

Reply:
> I have been pondering on this from: https://gitlab.com/CentOS/automotive/sample-images/-/merge_requests/451

It's important to understand this: The "signed OS image" model simply *cannot* have out of band arbitrary unsigned code injected - and that's exactly what this is.  It's state injected into a *disk image* that is unsigned.

Hence, this is why ultimately the model we need to get away from configuring disk images - and ensure that user desired mountpoints are created in the container (ostree commit).

Reply:
discussed the approach in meeting and we will keep the .service file in deployment stage with addition of the `create-mountpoint.service` for rhel9.4.